client/homebrew/pages/basePages/editorPage/editorPage.jsx

```diff
@@ -0,0 +1,593 @@
+/* eslint-disable max-lines */
+require('./editorPage.less');
+const React = require('react');
+const createClass = require('create-react-class');
+const _ = require('lodash');
+const request = require('superagent');
+const { Meta } = require('vitreum/headtags');
+
+const Nav = require('naturalcrit/nav/nav.jsx');
+const Navbar = require('../../../navbar/navbar.jsx');
+
+const NewBrew = require('../../../navbar/newbrew.navitem.jsx');
+const ReportIssue = require('../../../navbar/issue.navitem.jsx');
+const PrintLink = require('../../../navbar/print.navitem.jsx');
+const Account = require('../../../navbar/account.navitem.jsx');
+const RecentNavItem = require('../../../navbar/recent.navitem.jsx').both;
+
+const SplitPane = require('naturalcrit/splitPane/splitPane.jsx');
+const Editor = require('../../../editor/editor.jsx');
+const BrewRenderer = require('../../../brewRenderer/brewRenderer.jsx');
+
+const Markdown = require('naturalcrit/markdown.js');
+
+const googleDriveActive = require('../../../googleDrive.png');
+const googleDriveInactive = require('../../../googleDriveMono.png');
+
+const SAVE_TIMEOUT = 3000;
+
+const BREWKEY = 'homebrewery-new';
+const STYLEKEY = 'homebrewery-new-style';
+const METAKEY = 'homebrewery-new-meta';
+
+const EditorPage = createClass({
+	getDefaultProps : function() {
+		return {
+			brew : {
+				text      : '',
+				style     : '',
+				shareId   : null,
+				editId    : null,
+				createdAt : null,
+				updatedAt : null,
+				gDrive    : false,
+				trashed   : false,
+
+				title       : '',
+				description : '',
+				tags        : '',
+				published   : false,
+				authors     : [],
+				systems     : [],
+				renderer    : 'legacy'
+			},
```

Comment:
As I've become more familiar with React, I'm noticing more weird things with props and state that we have an opportunity to fix here. This section is one we can handle better, especially since I see that the related section in `getInitialState` is commented out.

So, props we know are passed in to components. `defaultProps` are what the component uses if nothing was passed in. A problem occurs however when the passed prop is an object with nested properties, like a brew with `title`, `text`, `renderer` and such. `getDefaultProps` only applies props that are fully missing, so if we pass in a brew here that is missing a few values (say an old brew where `renderer` or `style` isn't defined), it will only see that an object called `brew` exists and it won't supply any of the missing values.

This is why in `getInitialState` we had to break out each key in the brew and double-check that the value was there, and if not, set some default value. (continued in comment below)...